src/bluesky/plan_stubs.py

```diff
@@ -359,9 +364,11 @@ def rel_set(
     return (yield from relative_set_wrapper(abs_set(obj, *args, group=group, wait=wait, **kwargs)))
 
 
+# The format (device1, value1, device2, value2, ...)
+# is not currently able to be represented in python's type system
 @plan
 def mv(
-    *args: Tuple[Union[Movable, NamedMovable, Any], ...],
+    *args: Any,
```

Comment:
why not `Unpack[Tuple[Any]]` here as well?

Reply:
or why use the fancier spelling in `abs_set` / `rel_set`

Reply:
`*args: Unpack[Tuple[Any]]` means "one and only one positional arg of any type with any name", so that wouldn't work here, but `*args: Unpack[Tuple[Any, ...]]`, which would allow `args` to be a tuple of any types of any length, is the widest possible thing `args` can be in the language, and is equivalent to plain `Any` so I think that's more readable.

Reply:
@callumforrester @coretl any opinions on that?

Reply:
I like:
- typing follows the `bluesky.protocol`, and documents what we want people to use
- runtime allows for args and kwargs for backwards compat

So this really becomes: do we want to encourage the pattern of `set` taking a single argument?

If so, then I would prefer:
- `Movable[T]` having `def set(self, *args: Unpack[Tuple[T]])`
- `def abs_set(obj: Movable[T], *args: Unpack[Tuple[T]], ..., **kwargs)`
- `def mv(*args: Any, ...)` (possibly with some `typing.override`s for the 2, 4, 6 argument case like `asyncio.gather` has)
 
to encourage that close matching at the typing level. It still doesn't break any runtime things, but gives nice type hints when you try to `abs_set` a `float` Signal to a `str`.

Reply:
I actually have very few opinions: My original hope was to have a set of well-typed, backwards compatible plans that we could use in blueapi. For various reasons (not just this thread) that's not going to be possible and we will be writing wrapper plans with robust argument types. Something like this...

```python
Move[T] = Tuple[Movable[T], T]

def move(moves: list[Move[Any]], wait: bool = True) -> MsgGenerator:
    yield from bps.mv()

...
# in some plan...

yield from move((x, 2.0), (y, 3.0))
yield from move((x, 2.0), (y, "3.0"))  # fails type check
```

So the core bluesky typing should be as loose as is needed to avoid breaking backwards compatibility and maintain a smooth CLI/IPython shell experience.

Reply:
Ok, given the above (and us not having `Movable[T]` yet anyway) I'll make the signatures for `abs/rel_set` take anything and we can tighten it in our wrappers

Reply:
`list[Move[Any]]` is a much nicer API (from a programming point of view, not from a type-it-in-the-terminal point of view).

I would be happy for that to go upstream and `bps.mv` to be a shim that does pair-wise grouping and calls `move` (assuming we can come up with a reasonable way to not have `mv` and `move` both tab-complete in a cli.

Reply:
I just wanted to note that similar changes were made for built-in plans. For example type annotation for `scan` works well with the Queue Server:
https://github.com/bluesky/bluesky/blob/a40a025aef605174bc44980254c3a1ea8f95bfc9/src/bluesky/plans.py#L1189
I am adding support for `NamedMovable`, so the annotation can be extended. If the type is simply set to `Any`, it disables conversion of strings to ophyd objects, so it needs to be a union that includes `Movable` and/or `NamedMovable` for motors. Also `Any` can be replaced with something more specific. I will check this PR branch with QS before merging to make sure everything works.